src/main/java/com/company/library/CollectionTask.java

```diff
@@ -32,69 +39,119 @@ public static void main(String[] args) throws IOException, URISyntaxException {
 
-        }
-
-        return listMap;
+        }*/
+        return books.stream()
+                .collect(Collectors.groupingBy(book -> book.author));
     }
 
     //TODO change return type to appropriate one
     private static Collection<Author> findUniqueAuthors(List<Book> books) {
         //TODO return only unique authors
-        Set<Author> uniqueAuthors = new HashSet<>();
+        /*Set<Author> uniqueAuthors = new HashSet<>();
         for (Book book : books) {
             uniqueAuthors.add(book.author);
-        }
-
-        return uniqueAuthors;
-
+        }*/
+        return books.stream()
+                .map(book -> book.author)
+                .collect(Collectors.toSet());
     }
 
     //TODO change return type to appropriate one
     private static Collection<Book> findUniqueBooks(List<Book> books) {
         //TODO return only unique books
-        return new HashSet<>(books);
+        //return new HashSet<>(books);
+
+        return books.stream().distinct().collect(Collectors.toList());
+
     }
 
     public static List<String> getLinesFromFile(final String filepath) throws IOException, URISyntaxException {
-        return Files.lines(
-                Path.of(
-                        CollectionTask.class
-                                .getClassLoader()
-                                .getResource(filepath)
-                                .toURI()
-                )
-        )
-                .toList();
+
+        try {
+            return Files.lines(
+                    Path.of(
+                            CollectionTask.class
+                                    .getClassLoader()
+                                    .getResource(filepath)
+                                    .toURI()
+                    )
+            )
+                    .toList();
+        } catch (NullPointerException e) {
+            List<String> emptyFile = new ArrayList<>();
+            System.err.println("File doesn't found");
+            return emptyFile;
+        }
+
+
     }
 
+
     public static List<Book> instantiateBooks(List<String> lines) {
         //TODO parse string, instantiate Author and Book classes and return list of books
         List<Book> bookList = new ArrayList<>();
-        for (String e : lines) {
-            String[] columns = e.split(",");
-            bookList.add(new Book(new Author(columns[0], columns[1]), columns[2], Integer.parseInt(columns[3])));
+        try {
+
+            for (String e : lines) {
+                String[] columns = e.split(",");
+                bookList.add(new Book(new Author(columns[0], columns[1]), columns[2], Integer.parseInt(columns[3])));
+            }
+            return bookList;
+        } catch (NumberFormatException y) {
+
+            //Pattern exampleString = Pattern.compile(("[a-zA-Z],[a-zA-Z],[a-zA-Z],\\d{4}"), Pattern.CASE_INSENSITIVE);
+
+            System.err.println("Wrong data format at document");
+
         }
```

Comment:
This looks good to me! At least. it doesn't stop the app which is great.
However, this one works fine only for the case when invalid lines are at the end of the file.
What if they are in the middle? Or rotates with valid ones?